sycl/test-e2e/Matrix/element_wise_odd_sizes_bf16_impl.hpp

```diff
@@ -0,0 +1,107 @@
+#define SG_SZ 16
+
+static constexpr size_t M_MULTIPLIER = 16;
+
+static float make_fp32(bfloat16 x) {
+  unsigned int y = *((int *)&x);
+  y = y << 16;
+  float *res = reinterpret_cast<float *>(&y);
+  return *res;
+}
+
+template <typename T, size_t NUM_ROWS, size_t NUM_COLS> struct big_matrix {
+public:
+  T *mat;
+
+public:
+  T *get_data() { return mat; }
+  void set_data(T *data) { mat = data; }
+  big_matrix(T *data) : mat(data) {}
+};
+
+template <typename T, size_t M, size_t N>
+void assert_ops_ref(host_accessor<T, 2, access::mode::read> C,
+                    const float ref) {
+  for (size_t i = 0; i < M; i++)
+    for (size_t j = 0; j < N; j++) {
+      auto diff = make_fp32(C[i][j]) - ref;
+      assert(std::fabs(static_cast<float>(diff)) <
+             std::numeric_limits<float>::epsilon());
+    }
+}
+template <typename T, size_t M, size_t N, size_t TM, size_t TN, size_t TK>
```

Comment:
I see some confusion with M, N and K. If only matrix A is used, I suggest using only M and K, TM and TK.

Reply:
Done